End game gui implementation and subsequent fixes

hippi345 opened the pull request:
still needs the gui to be intuitive and the end game conditions to scale to multiple game scenarios

Commit:
just meant for cleaning up the movements

still needs to have a lot of other things reworked but wanted to PR out the moves action on manual clicks consolidated to a single method

Commit:
buggy

Commit:
started refactoring out redrawing the grid

stopped the infinite loop but end game GUI still not loading and game not closing - see Issue posted

Commit:
smallest change

Commit:
allow for stability on endgame gui while not breaking on autoplay

autoplay will die out on system exit and manual will bring up the endgame screen
long term we need both to work but this at least is less buggy and performance breaking

Commit:
fixed the bug with autoplay blocking gui and not showing the end game gui

still needs the end game gui fleshed out more

RiskRunner0 commented on src/sample/Constants.java:
```diff
@@ -7,4 +7,6 @@
     // colors
     static String LIGHT_BLUE = "-fx-background-color:#00BFFF; -fx-opacity:1;";
+
+    static  boolean GAME_COMPLETE = false;
```
Looks like 2 spaces

hippi345 replied:
wdym?

RiskRunner0 replied:
Between static and boolean, I think there's 2 spaces

hippi345 replied:
was separating those intentionally since they are different things but I am resolving

RiskRunner0 commented on src/sample/Game.java:
```diff
@@ -29,7 +30,7 @@ void InitializeGame(View gameView)
             this.bot.RandomizeLocation(gameView.gridSize);
         }
-        gameView.adjustBotAndTreasureLocations(this.bot, this.treasure);
+        // gameView.redrawGrid(this.bot, this.treasure);
```
If we don't need this, you can remove it.

hippi345 replied:
fairsies enough

RiskRunner0 commented on src/sample/Game.java:
```diff
@@ -77,51 +59,19 @@ private boolean treasureIsFound()
     {
-        if(treasureIsFound())
+        if(treasureIsFound()) {
+            Constants.GAME_COMPLETE = true;
```
Setting a constant is counter intuitive, if you're modifying it, it's not constant.  Maybe make this a field on the game?

hippi345 replied:
Autoplay being implemented in Main would make this bit involved but I could make it public and proceed on that?

RiskRunner0 replied:
Main is where the game is created, so you could do like ```Game.GameComplete``` in there, and then set that variable on the game here.

Commit:
Merge branch 'master' into endGameGUI

Commit:
Update Constants.java

hippi345 commented:
@QueenCitySmitty  could you give this one a run through with comments
I think some of the things considered needing fixed may have been addressed in one of your more recent PR's - so feel free to let me know of things in this PR that I can fix (short of needing a separate PR) as I don't know what has been addressed since this PR 2 days ago but I want to make sure it is sound to merge

RiskRunner0 commented:
I'm not seeing anything about the end game gui in this pr?

Commit:
committing

Commit:
Merge branch 'endGameGUI' of https://github.com/hippi345/randobotjava into endGameGUI

Commit:
bringing it all into the fold

Commit:
not buggy anymore - gonna re-implement end game gui

Commit:
not bugy and is working. going to PR this one but try to unblock threads on a microscale for the endgame gui being able to close the game gui next.

hippi345 commented:
@QueenCitySmitty 
Give this one a run and see if the execution is ok for a merge - I put the end game gui in the View class for sake of organization and implemented the endgame actions within your latest restructure for the Game so Game calls -> View to close the game and open the end game gui

Commit:
simplified - could not figure out GUI threaded stuff yet

RiskRunner0 commented on src/sample/newThreadItems.java:
```diff
@@ -0,0 +1,10 @@
+package sample;
+
+public class newThreadItems implements Runnable
```
Do we need this class right now for this PR?  I don't see it being used anywhere

hippi345 replied:
Resolved-this is in another commit external to this one.

RiskRunner0 commented on src/sample/Game.java:
```diff
@@ -86,6 +86,11 @@ private void executeTurn(MoveEnum botMovementDirection)
         Main.gameView.adjustBotAndTreasureLocations(previousBotPoint, this.bot, this.treasure);
+        if (_status == GameStatusEnum.Complete)
+        {
+            System.out.println("You found the treasure!");
```
We should probably expose status, and read from it after a move instead, we might need to use it later.

hippi345 replied:
wdym? 
for context, I did not make status and this change in the determination to end the game.
So I am not sure what you are asking on this comment. 
What do you mean by "expose"? Save it in memory? Output it to the console? Etc.

hippi345 replied:
@QueenCitySmitty

RiskRunner0 replied:
Expose it as make it a public field that anything can access, like at any time I should be able to query the game for the current status.

Commit:
Delete newThreadItems.java

Commit:
added a public getter

will make a task to see where getStatus() can be used in place of the current logic outside of Game

Commit:
Update Game.java
